internal/backends/sqlite/sqlite.go

```diff
@@ -48,13 +48,29 @@ type stats struct {
 }
 
 // collectionsStats returns statistics about tables and indexes for the given collections.
-func collectionsStats(ctx context.Context, db *fsql.DB, list []*metadata.Collection) (*stats, error) {
+//
+// If refresh is true, it calls ANALYZE on the tables of the given list of collections.
+//
+// If the list of collections is empty, then stats filled with zero values is returned.
+func collectionsStats(ctx context.Context, db *fsql.DB, list []*metadata.Collection, refresh bool) (*stats, error) {
+	if len(list) == 0 {
+		return new(stats), nil
+	}
+
 	var err error
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/3518
-	q := `ANALYZE`
-	if _, err = db.ExecContext(ctx, q); err != nil {
-		return nil, lazyerrors.Error(err)
+	if refresh {
+		queries := make([]string, len(list))
+		for i, c := range list {
+			queries[i] = fmt.Sprintf("ANALYZE %q;", c.TableName)
+		}
+
+		// SQLite ANALYZE does not allow multiple tables as arguments, hence
+		// build multiple queries such as `ANALYZE 'table1'; ANALYZE 'table2';`.
```

Comment:
@chilagrow are you sure that actually works? Is there a way to check that?

Reply:
I added tests for error handling.
- `ANALYZE 'table1'; ANALYZE 'table2';`
- `ANALYZE 'non-existent'; ANALYZE 'table2';`
- `ANALYZE 'table1'; ANALYZE 'non-existent';`

But we both also want to know that if `ANALYZE 'table1'` and `ANALYZE 'table2'` has been executed and updated stats. Unfortunately `SQLite` seems to have updated stats without calling 

`

Reply:
> I added tests for error handling and it seems to work

The problem with running `QUERY 1; QUERRY 2` in a single exec is that we don't know without going deep into `database/sql`'s driver code if it works correctly. Many drivers may run only the first or only the last statement. 

Let's run them one-by-one to be sure

Reply:
Okay, let me update it to send multiple queries :+1:

Reply:
@chilagrow edited ^

Reply:
Okay :+1: